extensions/gitpod-web/src/extension.ts

```diff
@@ -839,6 +836,16 @@ export function registerPorts(context: GitpodExtensionContext): void {
 			vscode.commands.executeCommand('gitpod.api.connectLocalApp', apiPort);
 		}
 	}));
+	vscode.workspace.onDidChangeConfiguration((e: vscode.ConfigurationChangeEvent) => {
+		if (!e.affectsConfiguration('gitpod.experimental.portsView.enabled')) {
+			return;
+		}
+		const isPortsViewExperimentEnable = vscode.workspace.getConfiguration('gitpod.experimental.portsView').get<boolean>('enabled');
+		vscode.commands.executeCommand('setContext', 'gitpod.portsView.visible', isPortsViewExperimentEnable);
```

Comment:
We use `setContext` since config will not reflect to changes sometimes.

`enabled=false` and change to `true` but `Ports` view not shown